crates/iota-indexer/src/main.rs

```diff
@@ -19,26 +58,180 @@ async fn main() -> Result<(), IndexerError> {
+
+            let rpc_client_url_parsed = old_conf
+                .rpc_client_url
+                .parse()
+                .expect("RPC Client url should be valid");
+
+            let command = if old_conf.analytical_worker {
+                Command::AnalyticalWorker
+            } else if old_conf.reset_db {
+                Command::ResetDatabase { force: true }
+            } else if old_conf.rpc_server_worker {
+                Command::JsonRpcService(JsonRpcConfig {
+                    iota_names_options: old_conf.iota_names_options,
+                    rpc_address: SocketAddr::new(
+                        old_conf
+                            .rpc_server_url
+                            .as_str()
+                            .parse()
+                            .expect("RPC Server url should be valid"),
+                        old_conf.rpc_server_port,
+                    ),
+                    rpc_client_url: old_conf.rpc_client_url,
+                })
+            } else {
+                Command::Indexer {
+                    ingestion_config: IngestionConfig {
+                        sources: IngestionSources {
+                            data_ingestion_path: old_conf.data_ingestion_path,
+                            remote_store_url: old_conf.remote_store_url.map(|url| {
+                                url.parse().expect("Remote Store URL should be correct")
+                            }),
+                            rpc_client_url: Some(rpc_client_url_parsed),
+                        },
+                        checkpoint_download_queue_size: download_queue_size,
+                        checkpoint_download_timeout: ingestion_reader_timeout_secs,
+                        checkpoint_download_queue_size_bytes: data_limit,
+                    },
+                    snapshot_config: SnapshotLagConfig {
+                        snapshot_min_lag,
+                        sleep_duration: SnapshotLagConfig::DEFAULT_SLEEP_DURATION_SEC,
+                    },
+                    pruning_options: PruningOptions {
+                        epochs_to_keep: std::env::var("EPOCHS_TO_KEEP")
+                            .map(|s| s.parse::<u64>().ok())
+                            .unwrap_or_else(|_e| None),
+                    },
+                }
+            };
+
+            iota_indexer::config::IndexerConfig {
+                database_url: db_url
+                    .map_err(|e| {
+                        IndexerError::PgPoolConnection(format!(
+                            "Failed parsing database url with error {e:?}"
+                        ))
+                    })?
+                    .expose_secret()
+                    .parse()
+                    .expect("Database URL should be correct"),
+                connection_pool_config: pool_config_from_env(),
+                metrics_address,
+                command,
+            }
+        }
+    };
```

Comment:
Failing to parse the new CLI may mean that `--help` is requested, or any other user-input error that might require feedback with printing help.

So falling back to the deprecated should be avoided. 

We should instead simplify things and first try to parse the deprecated config and on error fallback to the CLI allowing any error to be handled by `clap`. Something like

```rust
    let opts = match DeprecatedIndexerConfig::try_parse() {
        Ok(opts) => opts.try_into()?,
        Err(_) => IndexerConfig::parse(),
    };
```
with proper modifications to account for `version`.

This enforces the following behaviour:

* We support successful usage of the deprecated interface,
* Otherwise, we direct to usage of the new interface

Reply:
As implied by the previous comment it makes sense to implement `TryFrom<DeprecatedIndexerConfig> for IndexerConfig`.